### DISCUSSION TITLE
What are the best solutions to run local LLMs right now?

### DISCUSSION THREAD

**Discussion by silvestron** [oobabooga/text-generation-webui#2522] [General]
I'm a bit out of the loop, are 4bit GPTQ models still the best option for running local LLM on consumer hardware? I have an RTX 3060 with 12GB of VRAM.

I was also wondering if there have been developments for clients. Beside ooba I only know of TavernAI and its forks.

**Comment by BetaDoggo** [comment]
For running on gpu gptq is still the best. There are some new libraries for running gptq which are faster than gptq-for-llama like autogptq(requires a manual compilation for ooba) and exllama(Linux only for now and still a pr for ooba). GGML has gpu offloading support now which is much faster than gptq's offloading if you are trying to use a model that is larger than your vram.

For ui clients not much has happened. GPT4all has a decent chatgpt like client now with support for gpt-j models in addition to llama.

**Reply by silvestron** [reply]
Thank you! GGML can do GPU too now? That's crazy. I might take a look at it again, to run larger models.

Ooba's is still the best client I tried so far, it's only missing some QoL features that ChatGPT has like multiple chat history, inline editing and a "smart" continue feature that detects when a response is interrupted when it reaches the token limit.

**Reply by vdruts** [reply]
Exlamma works in windows and it's VERY fast. Try "magi" https://github.com/shinomakoi/magi_llm_gui

**Reply by ossirytk** [reply]
llama.cpp has support for gpu offloading now. You have to go through some hoops to enable it, but you can split the model between cpu and gpu. This enable you to run models that are too big for gpu and it's faster that running only on cpu. There are also some cpu perf improvements for cpu in the pipeline for llama.cpp right now.
